### PR TITLE
Some improvements on neo4j-core 7.0.0.alpha

### PR DESCRIPTION
**Pull request by ProGM**
This pull introduces/changes:
 * Some fixes on `CypherError` class and constructor

Pings:
@cheerfulstoic
@subvertallchris

### CODE REVIEW

File: spec/neo4j/core/cypher_session/cypher_error_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+      its(:original_message) { is_expected.to eq(message) }
+      its(:code) { is_expected.to eq(code) }
+      its(:stack_trace) { is_expected.to eq(stack_trace) }
+
+      %w(ConstraintValidationFailed ConstraintViolation).each do |error_code|
+        context "when passing a #{error_code}" do
+          let(:code) { error_code }
+          it { is_expected.to be_a(CypherSession::SchemaErrors::ConstraintValidationFailedError) }
+        end
```
Comment (cheerfulstoic) [inline spec/neo4j/core/cypher_session/cypher_error_spec.rb]: What about this?

```ruby
let_context code: 'ConstraintValidationFailed' do
  it { is_expected.to be_a(CypherSession::SchemaErrors::ConstraintValidationFailedError) }
end
let_context code: 'ConstraintViolation' do
  it { is_expected.to be_a(CypherSession::SchemaErrors::ConstraintValidationFailedError) }
end
```

**Comment by cheerfulstoic** [comment]
One comment, but otherwise this is great, thanks!

**Comment by ProGM** [comment]
Done 👍

**Comment by cheerfulstoic** [comment]
Released `alpha.2`, thanks!

**Pull request merged**